src/parser/expression.js

```diff
@@ -713,7 +713,8 @@ pp.parseObj = function (isPattern, refShorthandDefaultPos) {
       prop = this.parseSpread();
       prop.type = isPattern ? "RestProperty" : "SpreadProperty";
       node.properties.push(prop);
-      continue;
+      this.expect(tt.braceR);
```

Comment:
Is it easy to have have more specific error messages for these?

`let { ...x, y, z } = obj;` - spread has to be the last element (would result in the same value as the right hand side)
`let { x, ...y, ...z } = obj;` - cannot have multiple spreads (would result in the same value for both y and z)?